SimTracker/TrackTriggerAssociation/plugins/TTTrackAssociator.cc

```diff
@@ -86,91 +81,89 @@ void TTTrackAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, cons
         }  /// End of loop over the clusters
 
         /// Check if the stub is unknown
-        if (TTStubAssociationMapHandle->isUnknown(theseStubs.at(is)))
+        if (ttStubAssociationMapHandle->isUnknown(stub))
           ++mayCombinUnknown;
 
       }  /// End of loop over the stubs
 
-      /// If there more than 2 unknown stubs, go to the next track
+      /// If there are >= 2 unknown stubs, go to the next track
       /// as this track may be COMBINATORIC or UNKNOWN
+      /// (One unknown is allowed, if in 2S module).
       if (mayCombinUnknown >= 2)
         continue;
 
-      /// If we are here, all the stubs are either combinatoric or genuine
+      /// If we are here, all the stubs on track are either combinatoric or genuine
       /// and there is no more than one fake stub in the track
-      /// Loop over all the TrackingParticle which have been found in the track at some point
-      /// (stored in auxMap)
+      /// Loop over all the TrackingParticle which have been found in the track
+      /// (stored in auxMap), to check if any are present in all stubs on Track.
 
       std::vector<const TrackingParticle*> tpInAllStubs;
 
-      std::map<const TrackingParticle*, edm::Ptr<TrackingParticle>>::const_iterator iterAuxMap;
-      for (iterAuxMap = auxMap.begin(); iterAuxMap != auxMap.end(); ++iterAuxMap) {
-        /// Get all the stubs from this TrackingParticle
-        std::vector<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>, TTStub<Ref_Phase2TrackerDigi_>>>
-            tempStubs = TTStubAssociationMapHandle->findTTStubRefs(iterAuxMap->second);
+      for (const auto& auxPair : auxMap) {
+        /// Get all associated stubs of this TrackingParticle
+        std::vector<TTStubRef> tempStubs = ttStubAssociationMapHandle->findTTStubRefs(auxPair.second);
```

Comment:
capture by const ref

Reply:
Can't as TTStubAssociationMap::findTTStubRefs(TP) returns temporary vector if TP is not in map.